### PR TITLE
[WFLY-14864] Remove "always true" instanceof checks

### PR DESCRIPTION
**Pull request by boris-unckel**
Fixes https://issues.redhat.com/browse/WFLY-14864
Includes dead code removal.

### CODE REVIEW

File: ejb3/src/main/java/org/jboss/as/ejb3/component/stateless/StatelessComponentDescription.java [hunk 1]
```diff
@@ -191,9 +191,7 @@ public void configure(DeploymentPhaseContext context, ComponentConfiguration com
             }
         });
 
-
-        if (view instanceof EJBViewDescription) {
-            EJBViewDescription ejbViewDescription = (EJBViewDescription) view;
+            EJBViewDescription ejbViewDescription = view;
             if (ejbViewDescription.getMethodIntf() == MethodIntf.REMOTE) {
```
Comment (chengfang) [inline ejb3/src/main/java/org/jboss/as/ejb3/component/stateless/StatelessComponentDescription.java:195]: looks like we can directly use the variable `view`, without creating a new variable `ejbViewDescription`

File: ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/MessageDrivenComponentDescriptionFactory.java [hunk 2]
```diff
@@ -118,34 +118,17 @@ private void processMessageBeans(final DeploymentUnit deploymentUnit, final Coll
...
-                    messagingType = mdb.getMessagingType();
-                    final ActivationConfigMetaData activationConfigMetaData = mdb.getActivationConfig();
-                    if (activationConfigMetaData != null) {
-                        final ActivationConfigPropertiesMetaData propertiesMetaData = activationConfigMetaData.getActivationConfigProperties();
-                        if (propertiesMetaData != null) {
-                            for (final ActivationConfigPropertyMetaData propertyMetaData : propertiesMetaData) {
-                                activationConfigProperties.put(propertyMetaData.getKey(), propertyMetaData.getValue());
-                            }
+                final MessageDrivenBeanMetaData mdb = beanMetaData;
+                messagingType = mdb.getMessagingType();
```
Comment (chengfang) [inline ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/MessageDrivenComponentDescriptionFactory.java:122]: looks like we can directly use the existing variable `beanMetaData`, without creating the new local variable `mdb`.

**Comment by chengfang** [comment]
the check failed with an unused import `JBossGenericBeanMetaData`

File: ejb3/src/main/java/org/jboss/as/ejb3/component/stateless/StatelessComponentDescription.java (hunk 1 above)
Comment (boris-unckel) [reply ejb3/src/main/java/org/jboss/as/ejb3/component/stateless/StatelessComponentDescription.java:195]: Thank you for your fast reply. I have removed the unnecessary local variable.

File: ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/MessageDrivenComponentDescriptionFactory.java (hunk 2 above)
Comment (boris-unckel) [reply ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/MessageDrivenComponentDescriptionFactory.java:122]: Fixed.

**Review by chengfang — APPROVED** [review]

**Comment by boris-unckel** [comment]
@bstansberry Hello Brian, tests for this PR passed, PR is approved. Could you please check and merge?

**Pull request merged**